### ISSUE TITLE
Passing nesting to interpolated case doesn't work in v21.0.0.

### ISSUE THREAD

**Issue by Shin-Ogata** [i18next/i18next#1656]
## 🐛 Bug Report

Thank you for your great job.
It seems that [Passing nesting to interpolated](https://www.i18next.com/translation-function/nesting#passing-nesting-to-interpolated) case doesn't work in`v21.0.0`.

## To Reproduce

You can check it with the Jasmine test code below.
```js
it('check passing nesting to interpolated', async () => {
    const t = await i18next.init({
        lng: 'en-US',
        resources: {
            en: {
                translation: {
                    key1: 'hello world',
                    key2: 'say: {{val}}',
                },
            },
        },
    });
    expect(t('key2', { val: '$t(key1)' })).toBe('say: hello world');
    // -> say: $t(key1)
});
```

## Expected behavior

I have confirmed that it works correctly with `v20.6.1`.

## Your Environment

- *runtime version*: node v14
- *i18next version*: 21.0.0
- *os*: Windows

**Comment by adrai** [comment]
This is intended.
The default value for the skipOnVariables option was changed: https://www.i18next.com/misc/migration-guide#skiponvariables-true

**Comment by jamuhl** [comment]
Did you read the changelog? skipOnVariable?

**Comment by Shin-Ogata** [comment]
> Did you read the changelog? skipOnVariable?

Oh sorry, I didn't realize that this option affected this case.
It was written properly [here](https://www.i18next.com/translation-function/interpolation#all-interpolation-options),
and I set `skipOnVariable: false` and it worked.

Thank you for your quick answer.

**Comment by Shin-Ogata** [comment]
Just one more word.
In TypeScript, it seems that an error will occur if you try to specify `false`, but if it was `undefined`, it was OK.

```js
const t = await i18next.init({
    lng: 'en-US',
    resources: {
      :
    },
    interpolation: {
//      skipOnVariables: false,     // -> Type 'false' is not assignable to type 'true | undefined'.
        skipOnVariables: undefined, // works fine.
    },
});
```

**Comment by adrai** [comment]
Thank you. It has just been fixed in v21.0.1
